lib/health_cards/health_card.rb

```diff
@@ -1,214 +1,58 @@
-      def decompress_payload(payload)
-        inf = Zlib::Inflate.new(-Zlib::MAX_WBITS).inflate(payload)
-        JSON.parse(inf)
-      end
+    attr_reader :jws
 
-      # Compress an arbitrary payload, useful for debugging
-      # @param payload [Object] Any object that responds to to_s
-      # @return A compressed version of that payload parameter
-      def compress_payload(payload)
-        Zlib::Deflate.new(nil, -Zlib::MAX_WBITS).deflate(payload.to_s, Zlib::FINISH)
-      end
+    def_delegator :@jws, :verify
+    def_delegator :@qr_codes, :code_by_ordinal
+    def_delegators :@payload, :bundle, :issuer
 
-      # Sets/Gets the fhir version that will be passed through to the credential created by an instnace of
-      # this HealthCard (sub)class
-      # @param ver [String] FHIR Version supported by this HealthCard (sub)class. Leaving this param out
-      # will only return the current value
-      # value (used as a getter)
-      # @return [String] Current FHIR version supported
-      def fhir_version(ver = nil)
-        if @fhir_version.nil? && ver.nil?
-          @fhir_version = superclass.fhir_version unless self == HealthCards::HealthCard
-        elsif ver
-          @fhir_version = ver
-        end
-        @fhir_version
-      end
+    def initialize(jws)
+      @jws = jws.is_a?(String) ? JWS.from_jws(jws) : jws
+      @payload = Payload.from_payload(@jws.payload)
+      @qr_codes = QRCodes.from_jws(@jws)
     end
 
-    fhir_version '4.0.1'
-
-    additional_types 'https://smarthealth.cards#health-card'
-
-    allow type: FHIR::Meta, attributes: %w[security]
-
-    disallow attributes: %w[id text]
-    disallow type: FHIR::CodeableConcept, attributes: %w[text]
-    disallow type: FHIR::Coding, attributes: %w[display]
-
-    # Create a HealthCard
-    #
-    # @param bundle [FHIR::Bundle] VerifiableCredential containing a fhir bundle
-    # @param issuer [String] The url from the Issuer of the HealthCard
-    def initialize(bundle:, issuer: nil)
-      raise InvalidPayloadError unless bundle.is_a?(FHIR::Bundle) # && bundle.valid?
-
-      @issuer = issuer
-      @bundle = bundle
+    def credential
+      { verifiableCredential: [@jws.to_s] }
     end
 
-    # A Hash matching the VC structure specified by https://smarthealth.cards/#health-cards-are-encoded-as-compact-serialization-json-web-signatures-jws
-    # @param filter [Boolean] specifies whether the bundle should apply allow/disallow rules
-    #  and meta filtering features. Defaults to true.
-    # @return [Hash]
-    def to_hash(filter: true)
-      fhir_bundle = filter ? strip_fhir_bundle : bundle
-      {
-        iss: issuer,
-        nbf: Time.now.to_i,
-        vc: {
-          type: self.class.types,
-          credentialSubject: {
-            fhirVersion: self.class.fhir_version,
-            fhirBundle: fhir_bundle.to_hash
-          }
-        }
-      }
+    def to_json(*_args)
+      credential.to_json
```

Comment:
If we change `credential` we'll need to update the name here too (and in other places)